yt/geometry/geometry_enum.py

```diff
@@ -0,0 +1,42 @@
+import sys
+from enum import auto
+
+if sys.version_info >= (3, 11):
+    from enum import StrEnum
+else:
+    from yt._maintenance.backports import StrEnum
+
+# register all valid geometries
+class Geometry(StrEnum):
+    CARTESIAN = auto()
+    CYLINDRICAL = auto()
+    POLAR = auto()
+    SPHERICAL = auto()
+    GEOGRAPHIC = auto()
+    INTERNAL_GEOGRAPHIC = auto()
+    SPECTRAL_CUBE = auto()
+
+    def __str__(self):
+        # Implemented for backward compatibility.
+        if sys.version_info >= (3, 11):
+            return super().__str__()
+        else:
+            return self.name.lower()
+
+    def from_str(_input: str, /) -> "Geometry":
+        """Convert a str to a Geometry enum member in a case-insensitve manner.
+
+        Raises
+        ------
+
+        ValueError in case of unrecognized geometry
+        """
+        str_to_member = {str(member): member for member in Geometry}
+        try:
+            return str_to_member[_input.lower()]
+        except KeyError:
+            raise ValueError(f"Unknown geometry {_input!r}") from None
+
+    def is_curvilinear(self) -> bool:
+        # TODO(4244): geographic geometries should probably be considerd curvilinear too ?
```

Comment:
I guess that would be a question for @chrishavlin

Reply:
Nevermind I'm just deprecating the function altogether because in hindsight I think it wasn't designed with geographic geometries in mind.